python_modules/dagster/dagster/_core/host_representation/grpc_server_registry.py

```diff
@@ -268,11 +281,23 @@ def __exit__(self, exception_type, exception_value, traceback):
             cast(threading.Event, self._cleanup_thread_shutdown_event).set()
             self._cleanup_thread.join()
 
+        self.shutdown_all_processes()
+
+        if self._wait_for_processes_on_shutdown:
+            self.wait_for_processes()
+
+    def shutdown_all_processes(self):
         for process in self._all_processes:
             process.shutdown_server()
 
-        if self.instance.code_server_settings.get("wait_for_local_processes_on_shutdown", False):
-            self.wait_for_processes()
+    def have_all_servers_shut_down(self) -> bool:
```

Comment:
nit: i got confused by the name interpreting it as a command "have all servers shut down" instead of a question. Consider `all_servers_have_shutdown`, `check_...` or something